platform/core/PolyPodCoreAndroid/core/src/main/kotlin/coop/polypoly/core/Core.kt

```diff
@@ -3,80 +3,34 @@ package coop.polypoly.core
 import org.msgpack.core.MessagePack
 import org.msgpack.core.MessageUnpacker
 import org.msgpack.value.Value
-import org.msgpack.value.ValueFactory
 
 class Core {
     companion object {
-        fun bootstrapCore(languageCode: String, fsRoot: String) {
+        fun bootstrapCore(args: BootstrapArgs) {
             return handleCoreResponse(
                 JniApi.bootstrapCore(
-                    languageCode,
-                    fsRoot,
+                    args.asValue().pack(),
                     JniApi
                 )
             ) {}
         }
 
-        fun loadFeatureCategories(
-            featuresDir: String,
-            forceShow: List<FeatureCategoryId>
-        ): List<FeatureCategory> {
-            val args = mutableMapOf<Value, Value>()
-            args[ValueFactory.newString("features_dir")] =
-                ValueFactory.newString(featuresDir)
-            args[ValueFactory.newString("force_show")] =
-                ValueFactory.newArray(
-                    forceShow.map { ValueFactory.newString(it.toString()) }
-                )
-            val bytes = ValueFactory.newMap(args).pack()
-            return handleCoreResponse(JniApi.loadFeatureCategories(bytes)) {
-                mapFeatureCategories(
-                    it
-                )
-            }
-        }
-
-        fun execRdfQuery(query: String): Value {
-            return handleCoreResponse(
-                JniApi.execRdfQuery(query)
-            ) { it }
-        }
-
-        fun execRdfUpdate(query: String) {
+        fun <CoreResponse> executeRequest(
+            request: CoreRequest,
+            // TODO: Investigate the option of doing automatic decoding instead of asking for a decoder.
```

Comment:
Even better, automatic conversion from Message Pack bytes to a given CoreResponse. Developers will just have to create a model which conforms to some deserialisation interface. It could be something like this:
```kotlin
@Deserializable
data class Feature {...}
```

In the same way we have automatic encoding on iOS, the models just have to conform to `Encodable` protocol, nothing more.